core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1387,16 +1387,18 @@ private[spark] object Utils extends Logging {
           catchBlock
         } catch {
           case t: Throwable =>
-            originalThrowable.addSuppressed(t)
-            logWarning(s"Suppressing exception in catch: " + t.getMessage, t)
+            if (originalThrowable != t) {
+              originalThrowable.addSuppressed(t)
+              logWarning(s"Suppressing exception in catch: " + t.getMessage, t)
```

Comment:
@devaraj-kavali this is fine to merge but I noticed an existing minor problem in the message here: should be `s"Suppressing ... ${t.getMessage}"`